internal/controllers/core/extensionrepo/reconciler.go

```diff
@@ -230,8 +230,11 @@ func (r *Reconciler) reconcileDownloaderRepo(ctx context.Context, state *repoSta
 	state.lastFetch = time.Now()
 
 	needsDownload := true
-	if exists && state.spec.Ref != "" {
-		// If an explicit ref is specified, there's no reason to pull a new version.
+	if exists && state.spec.Ref != "" && state.spec.Ref != "HEAD" {
```

Comment:
i'm actually not totally sure what the right API is here for tags that can change server-side...but this seems like a good stopgap until we figure out what we want.

Reply:
Agreed this is probably a reasonable stop-gap, and custom extension repos can still take advantage of rolling updates by ensuring their default branch is `stable` so that it'd be `HEAD`, right?

Longer-term, we could probably inspect the _local_ ref to determine the type, and if it doesn't exist yet or is a branch, we could fetch updates but if it exists & is a tag/commit SHA, skip the sync.

Reply:
fwiw, i think the correct behavior here is to do something like:
- always fetch the remote origin
- resolve the `Ref` in the remote origin to a git hash
- checkout the git hash
but it also feels like a bit of a rabbithole that's not worth spending time optimizing